File: Data Structures & Algorithms/duplicate-integer/submission-1.cpp
class Solution {
public:
    bool hasDuplicate(vector<int>& nums) {
        unordered_map<int,int> hashSet;
        for(int it:nums){
            if(hashSet[it]==1) return true;
            hashSet[it] +=1;
            
        }
        return false;
    }
};

File: Data Structures & Algorithms/search-insert-position/submission-1.py
class Solution:
    def searchInsert(self, nums: List[int], target: int) -> int:
        left = 0
        right  = len(nums)-1
        mid = left + (right-left)//2
        while(left<=right):
            if nums[mid]==target:
                return mid
            elif target>nums[mid]:
                left = mid+1
            else:
                right = mid-1
            mid = left+(right-left)//2
        return left